[Bug]: @storybook/core is (wrongly) specifying @storybook/global as a devDependency

### Describe the bug

Storybook 8.2 (re-)introduces the `@storybook/core` dependency. That package imports from `@storybook/global`, but specifies it only as a `devDependency`. This works just fine in most setups because various other storybook packages do specify `@storybook/global` as a dependency, so it gets installed anyway.

However, when using yarn berry with [Plug'n'Play](https://yarnpkg.com/features/pnp), a package can only access other packages if they are specified in the dependency list.
There's a workaround using [packageExtensions](https://yarnpkg.com/configuration/yarnrc#packageExtensions) in the `.yarnrc.yml`, but IMO the proper solution would be to upgrade `@storybook/global` to a proper dependency of `@storybook/core`.

### Reproduction link

probably not necessary for this case

### Reproduction steps

_No response_

### System

```bash
Shouldn't be relevant to the problem at hand, can be provided if necessary.
```


### Additional context

_No response_

@shilman Storybook itself seems to run fine. When running tests with jest (and using `composeStories`), I get the following error message:

    Cannot find module '@storybook/global' from '/home/alex/local/path/to/repo/.yarn/cache/@storybook-core-npm-8.2.0-bd0bae05cc-bdc4f37182.zip/node_modules/@storybook/core/dist/components'

Adding the following to the `.yarnrc.yml` solves the issue:

     packageExtensions:
       "@storybook/core@*":
          dependencies:
            "@storybook/global": "*"

So does disabling Plug'n'Play.

I haven't used `tsup` yet, so I'm not sure what to expect. What I can tell is that I'm seeing imports from `@storybook/global` in `dist/components/index.js` and `dist/components.index.cjs`.

--- a/code/core/scripts/entries.ts
+++ b/code/core/scripts/entries.ts
@@ -28,12 +28,7 @@ export const getEntries = (cwd: string) => {
     define('src/preview-api/index.ts', ['browser', 'node'], true),
     define('src/manager-api/index.ts', ['browser', 'node'], true, ['react']),
     define('src/router/index.ts', ['browser', 'node'], true, ['react']),
-    define('src/components/index.ts', ['browser', 'node'], true, [
-      'react',
-      'react-dom',
-      '@storybook/csf',
-      '@storybook/global',
-    ]),
+    define('src/components/index.ts', ['browser', 'node'], true, ['react', 'react-dom']),
     define('src/theming/index.ts', ['browser', 'node'], true, ['react']),
     define('src/theming/create.ts', ['browser', 'node'], true, ['react']),
     define('src/docs-tools/index.ts', ['browser', 'node'], true),

